### PR TITLE
feat: add public squad about step in checklist

### PR DESCRIPTION
**Pull request by ilasw**
## Changes
- Added `Learn about Squads` step in squad's checklist for admin;
- Moved checks on step visibility inside the single steps using `condition` optional property;
- Fixed wrong label adding `Approved` label for public request.

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

### Preview domain
https://mi-491-squad-checklist.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/components/checklist/PublicAboutStep.tsx
Comment (nensidosari) [inline packages/shared/src/components/checklist/PublicAboutStep.tsx:10]: AboutPublicSquadStep

File: packages/shared/src/components/checklist/PublicAboutStep.tsx
Comment (nensidosari) [inline packages/shared/src/components/checklist/PublicAboutStep.tsx:16]:
```suggestion
    completeAction(ActionType.HideAboutPublicSquad);
```

File: packages/shared/src/components/squads/PublicSquadSubmissionActions.tsx
Comment (nensidosari) [inline packages/shared/src/components/squads/PublicSquadSubmissionActions.tsx:85]: Is this used in the checklist? I think the idea is to remove this step from checklist, but now that I see it, not sure if we have considered what will happen with all squads that have requested but we haven't approved yet? before we release this feature?

File: packages/shared/src/lib/checklist.ts
Comment (nensidosari) [inline packages/shared/src/lib/checklist.ts:56]: this ticket seem to include removing this step as well

File: packages/shared/src/graphql/actions.ts
Comment (nensidosari) [inline packages/shared/src/graphql/actions.ts:29]: Now this does not hide the step rather it marks it completed right?

**Review by nensidosari — COMMENTED** [review]
Some comments to check but otherwise looks good

File: packages/shared/src/hooks/useSquadChecklist.tsx
Comment (rebelchris) [inline packages/shared/src/hooks/useSquadChecklist.tsx:171]: Should also only be for admins? (not sure if this whole section maybe is already)

**Review by rebelchris — CHANGES_REQUESTED** [review]
Besides Nensi's comment only 1 thing

File: packages/shared/src/hooks/useSquadChecklist.tsx
Comment (ilasw) [reply packages/shared/src/hooks/useSquadChecklist.tsx:171]: I put this step only in the list for admins ([you can check here](https://github.com/dailydotdev/apps/pull/3489/files#diff-c6a0631a6e91d8c7c9e15687f2859e1614603ee0c6a48a00fd38de37166e0eb3R51)), so shouldn't be visible for users ✅

Comment (rebelchris) [reply packages/shared/src/hooks/useSquadChecklist.tsx:171]: Ah perfect then :D

**Review by rebelchris — APPROVED** [review]
Good from my end, but look at Nensi comments as well :D

File: packages/shared/src/graphql/actions.ts
Comment (ilasw) [reply packages/shared/src/graphql/actions.ts:29]: Actually is marked as done, I'm changing the name to reflect the behaviour! Thanks ✅

File: packages/shared/src/components/squads/PublicSquadSubmissionActions.tsx
Comment (ilasw) [reply packages/shared/src/components/squads/PublicSquadSubmissionActions.tsx:85]: Currently:
If the squad _is public_: we show "Learn about public squads"
If the squad _is private_: we show "Get more traffic by going public"

Is that right?

File: packages/shared/src/components/checklist/PublicAboutStep.tsx
Comment (ilasw) [reply packages/shared/src/components/checklist/PublicAboutStep.tsx:16]: Changed!

File: packages/shared/src/components/squads/PublicSquadSubmissionActions.tsx
Comment (sshanzel) [reply packages/shared/src/components/squads/PublicSquadSubmissionActions.tsx:85]: We should not need this item anymore: https://dailydotdev.atlassian.net/wiki/spaces/HAN/pages/1083408386/RFC-058+-+Public+Squads+GA#Public-Squad-eligibility

Comment (sshanzel) [reply packages/shared/src/components/squads/PublicSquadSubmissionActions.tsx:85]: We should not need any approvals anymore. Any rejected Squads in the past, shouldn't matter if they try to turn public since we won't surface public Squads anymore unless you are a member or the Squad is featured.

File: packages/shared/src/hooks/useSquadChecklist.tsx
Comment (sshanzel) [inline packages/shared/src/hooks/useSquadChecklist.tsx:160]: We should completely remove this item: https://dailydotdev.atlassian.net/wiki/spaces/HAN/pages/1083408386/RFC-058+-+Public+Squads+GA#Public-Squad-eligibility

**Review by sshanzel — CHANGES_REQUESTED** [review]
We should remove the going public item. After that, we can probably avoid the changes about the `condition` property.

File: packages/shared/src/components/squads/PublicSquadSubmissionActions.tsx
Comment (ilasw) [reply packages/shared/src/components/squads/PublicSquadSubmissionActions.tsx:85]: @sshanzel Updated, let me know if now is ok. "Learn about squads" is always visible for admins, but once you click on "Learn more" will be marked as completed for all squads.

Comment (sshanzel) [reply packages/shared/src/components/squads/PublicSquadSubmissionActions.tsx:85]: Yes, once clicked, it should be completed for all Squads.

**Review by sshanzel — APPROVED** [review]
Looks great! However, we should not merge it yet. We will do it at the same time we merge the create/edit page PRs. They are ready, just waiting the go signal from the platform team.

**Review by nensidosari — APPROVED** [review]

File: packages/shared/src/components/squads/PublicSquadSubmissionActions.tsx
Comment (nensidosari) [reply packages/shared/src/components/squads/PublicSquadSubmissionActions.tsx:85]: @ilasw are we removing this thing now based on Lee's comment above? https://github.com/dailydotdev/apps/pull/3489/files#r1738285343

**Review by nensidosari — CHANGES_REQUESTED** [review]
There seem to be still things we need to remove, see my comments above

**Comment by ilasw** [comment]
@nensidosari in this branch the `PublicSquadSubmissionActions` component is still used in PublicSquadEligibilityCard and in settings page, I cannot remove it. I suppose that this should be done in the PR that remove those component, wdyt?

**Comment by sshanzel** [comment]
> @nensidosari in this branch the `PublicSquadSubmissionActions` component is still used in PublicSquadEligibilityCard and in settings page, I cannot remove it. I suppose that this should be done in the PR that remove those component, wdyt?

In the other PR, I've already removed it. We'd really have to wait for the platform side of things since we removed the eligibility item from this PR. Otherwise, we will create a new PR to just add the new item.

**Comment by sshanzel** [comment]
I now pointed this branch to the intermediate one. Feel free to update the concerns.

**Review by nensidosari — APPROVED** [review]
That's fine, at least we have a intermediary branch now in case we don't want to merge directly

**Pull request merged**
